workitem/link/link_repository.go

```diff
@@ -195,31 +201,35 @@ func (r *GormWorkItemLinkRepository) List(ctx context.Context) (*app.WorkItemLin
 
 // Delete deletes the work item link with the given id
 // returns NotFoundError or InternalError
-func (r *GormWorkItemLinkRepository) Delete(ctx context.Context, ID satoriuuid.UUID) error {
-	var link = WorkItemLink{
-		ID: ID,
-	}
+func (r *GormWorkItemLinkRepository) Delete(ctx context.Context, linkID satoriuuid.UUID, suppressorID satoriuuid.UUID) error {
+	var l = WorkItemLink{}
+	tx := r.db.Where("id = ?", linkID).Find(&l)
 	log.Info(ctx, map[string]interface{}{
-		"wilID": ID,
+		"wilID": linkID,
 	}, "Deleting the work item link repository")
 
-	db := r.db.Delete(&link)
-	if db.Error != nil {
+	tx.Delete(&l)
+	if tx.RowsAffected == 0 {
+		return errors.NewNotFoundError("work item link", linkID.String())
+	}
+	if tx.Error != nil {
 		log.Error(ctx, map[string]interface{}{
-			"wilID": ID,
-			"err":   db.Error,
-		}, "unable to delete work item link repository")
-		return errors.NewInternalError(db.Error.Error())
+			"wilID": linkID,
+			"err":   tx.Error,
+		}, "unable to delete work item link")
+		return errors.NewInternalError(tx.Error.Error())
 	}
-	if db.RowsAffected == 0 {
-		return errors.NewNotFoundError("work item link", ID.String())
+	// save a revision of the deleted work item link
+	if err := r.revisionRepo.Create(ctx, suppressorID, RevisionTypeDelete, l); err != nil {
+		return errs.WithStack(err)
```

Comment:
Please use `errs.Wrapf(err, "")` for newly added errors like this to give some context.